### DISCUSSION TITLE
Allow other grant_types in gatsby-source-wordpress (wpcom integration)

### DISCUSSION THREAD

**Discussion by robations** [gatsbyjs/gatsby#24875] [Ideas / Feature Requests]
## Summary

`gatsby-source-wordpress` only allows the "password" grant type for integrating with wordpress.com, suggest we allow to configure `authorization_code` grant type as well.

### Basic example

Allow to specify an authorization code as [documented here](https://developer.wordpress.com/docs/oauth2/) using the `authorization_code` grant type.

```
auth: {
    wpcom_app_clientSecret: "xxxyyyzzz",
    wpcom_app_clientId: "999999",

    // implies authorization_code grant type:
    wpcom_app_code: "abcdefg123",
}
```

### Motivation

This would allow:

- Better security (don't use your Wordpress.com a/c username and password outside your direct control)
- Other authn methods within Wordpress.com (sign in with Google, sign in with Apple)

**Comment by TylerBarnes** [comment]
Hi @robations , thanks for opening a feature request!
I believe we wont be adding this because we're soon deprecating our usage of the WP REST API in favour of using WPGraphQL (see this [temporary repo](https://github.com/gatsbyjs/gatsby-source-wordpress-experimental)). As far as I'm aware because of this we will be unable to support WordPress.com instances. It's been my understanding that WPGraphQL doesn't work with wp.com but I did just look into it a bit and it appears you can upload custom plugins so the new plugin & WPGraphQL may work with wp.com https://wordpress.com/support/plugins/adding-plugins/#install-plugins-via-a-zip-file

@jasonbahl do you have any knowledge about using WPGraphQL and wp.com?

If WPGraphQL and the new version of `gatsby-source-wordpress` can work with wp.com, then we can think about adding this feature. If that is the case please open an issue [here](https://github.com/gatsbyjs/gatsby-source-wordpress-experimental), otherwise we'll have to unfortunately close this.

Thanks!
